src/AudioWaveformer.cpp

```diff
@@ -47,10 +47,20 @@ AudioWaveformData AudioWaveformer::ExtractSamples(int channel, int num_per_secon
         int total_samples = num_per_second * (reader->info.duration + 1.0);
         int extracted_index = 0;
 
+        // Force output to zero elements for non-audio readers
+        if (!reader->info.has_audio) {
+            total_samples = 0;
```

Comment:
Images (which have no audio data), also include a really long duration (60 * 60 * 30), and we need that to be zeroed out, otherwise that could generate a really long array of zeros.